tests/lib/nested.sh

```diff
@@ -19,11 +19,42 @@ NESTED_SSH_PORT=8022
 NESTED_MON_PORT=8888
 
 nested_wait_for_ssh() {
-    local retry wait
-    retry=400
-    wait=1
+    nested_retry_until_success 400 1 "true"
+}
+
+nested_wait_for_no_ssh() {
+    nested_retry_while_success 200 1 "true"
+}
+
+nested_get_boot_id() {
+    nested_retry_until_success 200 1 "cat /proc/sys/kernel/random/boot_id"
+}
 
-    while ! nested_exec true; do
+nested_wait_for_reboot() {
+    local initial_boot_id="$1"
+    local retry wait last_boot_id
+    retry=120
+    wait=5
+
+    last_boot_id=""
+    while [ $retry -ge 0 ]; do
```

Comment:
mmm so in the worst case that we can't ssh to the machine (which is all too common unfortunately :frowning: ), this code results in waiting for a total of `120*(200*1 + 5) = 24600` seconds or almost 7 hours, which I think is a tad too long for this kind of timeout, can we instead something fail quicker?

Reply:
yes, I'll try.

Reply:
@anonymouse64 I just pushed a fix for that. Thanks for the review